server/embed/etcd.go

```diff
@@ -227,15 +227,14 @@ func StartEtcd(inCfg *Config) (e *Etcd, err error) {
 
 	if srvcfg.ExperimentalEnableDistributedTracing {
 		tctx := context.Background()
-		tracingExporter, opts, err := setupTracingExporter(tctx, cfg)
+		tracingExporter, err := newTracingExporter(tctx, cfg)
 		if err != nil {
 			return e, err
 		}
-		if tracingExporter == nil || len(opts) == 0 {
-			return e, fmt.Errorf("error setting up distributed tracing")
```

Comment:
You intentionally removed this, does it mean that `tracingExporter == nil || len(opts) == 0` will never be true if the returned err is `nil`?

Reply:
Yes, that's the intention. I noticed no code path would result in err == nil and either exporter nor opts would be nil

Reply:
ack